integration_tests/src/main/python/aqe_test.py

```diff
@@ -128,6 +128,55 @@ def do_join(spark):
     assert_gpu_and_cpu_are_equal_collect(do_join, conf=_adaptive_conf)
 
 
+@allow_non_gpu("ProjectExec")
+def test_aqe_broadcast_join_non_columnar_child(spark_tmp_path):
+    data_path = spark_tmp_path + '/PARQUET_DATA'
+    def prep(spark):
+        data = [
+            (("Adam ", "", "Green"), "1", "M", 1000, "http://widgets.net"),
+            (("Bob ", "Middle", "Green"), "2", "M", 2000, "http://widgets.org"),
+            (("Cathy ", "", "Green"), "3", "F", 3000, "http://widgets.net")
+        ]
+        schema = (StructType()
+                  .add("name", StructType()
+                       .add("firstname", StringType())
+                       .add("middlename", StringType())
+                       .add("lastname", StringType()))
+                  .add("id", StringType())
+                  .add("gender", StringType())
+                  .add("salary", IntegerType())
+                  .add("website", StringType()))
+
+        df = spark.createDataFrame(spark.sparkContext.parallelize(data),schema)
+        df2 = df.withColumn("dt",current_date().alias("dt")).withColumn("ts",current_timestamp().alias("ts"))
+
+        df2.write.format("parquet").mode("overwrite").save(data_path)
+
+    with_cpu_session(prep)
+
+    def do_it(spark):
+        newdf2 = spark.read.parquet(data_path)
+        newdf2.createOrReplaceTempView("df2")
+
+        return spark.sql(
+            """
+            select 
+                a.name.firstname,
+                a.name.lastname,
+                b.full_url
+            from df2 a join (select id, concat(website,'/path') as full_url from df2) b
+                on a.id = b.id
+            """
+        )
+
+    conf = copy_and_update({ 'spark.rapids.sql.expression.Concat': 'false' }, _adaptive_conf)
```

Comment:
it does not make a semantic difference but `copy_and_update` is meant to create a clone of the conf object you want to preserve (first arg) with the update from the patch (second arg). So for consistency should be:
  ```suggestion
    conf = copy_and_update(_adaptive_conf, {'spark.rapids.sql.expression.Concat': 'false'})
```